core/src/test/java/org/apache/iceberg/util/TestParallelIterable.java

```diff
@@ -40,18 +40,41 @@ public class TestParallelIterable {
   public void closeParallelIteratorWithoutCompleteIteration()
       throws IOException, IllegalAccessException, NoSuchFieldException {
     ExecutorService executor = Executors.newFixedThreadPool(1);
+    Iterator<Integer> integerIterator =
```

Comment:
I would suggest to add a new test method rather than modifying an existing test in order to reproduce the issue